python/feldera/rest/feldera_client.py

```diff
@@ -77,16 +77,17 @@ def get_program(self, name: str, with_code: bool = False) -> Program:
             version=resp.get("version"),
         )
 
-    def compile_program(self, program: Program):
+    def compile_program(self, program: Program, config: Optional[dict] = None):
         """
         Compiles a SQL program
 
         :param program: The program to compile
+        :param config: Optional. The configuration for the compilation
         """
         body = {
             "code": program.code,
             "description": program.description or "",
-            "config": { "profile": "optimized" }
+            "config": config or {}

```

Comment:
Default without config can still have profile optimized in it (minor comment). Just for consistency across the Python API that optimized is the default profile.